.gitignore

```diff
@@ -35,6 +35,7 @@ Vagrantfile.local
 config/server-*
 config/zookeeper-*
 core/data/*
+core/data2/*
```

Comment:
perhaps add a note that this is created by a test [1], other wise other reviewers may wonder (like me) as to where is this directory coming from

https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala#L2673

Reply:
@divijvaidya  Yes it comes from this test